### PR TITLE
ONNX v2 raises an Exception when using PyTorch < 1.8.0 

### PR DESCRIPTION
**Pull request by mfuntowicz**

### CODE REVIEW

**Comment by mfuntowicz** [comment]
@sgugger failing tests seems unrelated to this PR, let you check 👍🏻

File: src/transformers/file_utils.py [hunk 1]
```diff
@@ -297,10 +297,10 @@ def is_torch_cuda_available():
...
 
-_torch_fx_available = False
+_torch_fx_available = _torch_onnx_dict_inputs_support_available = False
 if _torch_available:
     torch_version = version.parse(importlib_metadata.version("torch"))
-    _torch_fx_available = (torch_version.major, torch_version.minor) == (
+    _torch_fx_available = _torch_onnx_dict_inputs_support_available = (torch_version.major, torch_version.minor) == (
         TORCH_FX_REQUIRED_VERSION.major,
         TORCH_FX_REQUIRED_VERSION.minor,
     )
```
Comment (sgugger) [inline src/transformers/file_utils.py:306]: We should have a different check as torch_fx requires PyTorch 1.8 exactly and `_torch_onnx_dict_inputs_support_available` requires PyTorch 1.8 or greater.

File: src/transformers/file_utils.py [hunk 2]
```diff
@@ -300,11 +300,16 @@ def is_torch_cuda_available():
...
-    _torch_fx_available = _torch_onnx_dict_inputs_support_available = (torch_version.major, torch_version.minor) == (
+    _torch_fx_available = (torch_version.major, torch_version.minor) == (
         TORCH_FX_REQUIRED_VERSION.major,
         TORCH_FX_REQUIRED_VERSION.minor,
     )
 
+    _torch_onnx_dict_inputs_support_available = (torch_version.major, torch_version.minor) >= (
+        TORCH_ONNX_DICT_INPUTS_MINIMUM_VERSION.major,
+        TORCH_ONNX_DICT_INPUTS_MINIMUM_VERSION.minor
+    )
```
Comment (sgugger) [inline src/transformers/file_utils.py]:
```suggestion
    _torch_onnx_dict_inputs_support_available = torch_version >= TORCH_ONNX_DICT_INPUTS_MINIMUM_VERSION
```
You can directly compare the parsed version :-)

File: src/transformers/file_utils.py [hunk 3]
```diff
@@ -275,7 +275,7 @@
 }
 
 # This is the version of torch required to run torch.fx features and torch.onnx with dictionary inputs.
-TORCH_FX_REQUIRED_VERSION = version.parse("1.8")
+TORCH_FX_REQUIRED_VERSION = TORCH_ONNX_DICT_INPUTS_MINIMUM_VERSION = version.parse("1.8")
```
Comment (sgugger) [inline src/transformers/file_utils.py:278]: Let's set this on two lines, just in case one or the other change later.

**Review by sgugger — APPROVED** [review]
A few more nits and we should be good to go!

File: src/transformers/onnx/convert.py [hunk 4]
```diff
@@ -79,7 +80,10 @@ def export(
 
     """
     if not is_torch_available():
-        raise Exception("Cannot convert because PyTorch is not installed. Please install torch first.")
+        raise ImportError("Cannot convert because PyTorch is not installed. Please install torch first.")
+
+    if not is_torch_onnx_dict_inputs_support_available():
+        raise AssertionError(f"Unsupported PyTorch version, minimum required is 1.8.0, got: {torch_version}")
```
Comment (sgugger) [inline src/transformers/onnx/convert.py:86]: You should import torch_version here once we know torch is available

**Pull request merged**